Gems/ROS2/Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,485 @@
+                    saveOutcome.GetError().c_str());
+                return;
+            }
+
+            // Add asset to expected assets list
+            if (assetFound)
+            {
+                AZStd::lock_guard lock{ m_buildMutex };
+                m_meshesToBuild.push_back(AZ::IO::Path(assetInfo.m_relativePath));
+            }
+        }
+    }
+
+    void CollidersMaker::AddColliders(urdf::LinkSharedPtr link, AZ::EntityId entityId)
+    {
+        AZStd::string typeString = "collider";
+        const bool isWheelEntity = Utils::IsWheelURDFHeuristics(link);
+        if (isWheelEntity)
+        {
+            AZ_Printf(Internal::collidersMakerLoggingTag, "Due to its name, %s is considered a wheel entity", link->name.c_str());
+        }
+        const AZ::Data::Asset<Physics::MaterialAsset> materialAsset =
+            isWheelEntity ? m_wheelMaterial : AZ::Data::Asset<Physics::MaterialAsset>();
+        size_t nameSuffixIndex = 0; // For disambiguation when multiple unnamed colliders are present. The order does not matter here
+        for (auto collider : link->collision_array)
+        { // one or more colliders - the array is used
+            AddCollider(
+                collider, entityId, PrefabMakerUtils::MakeEntityName(link->name.c_str(), typeString, nameSuffixIndex), materialAsset);
+            nameSuffixIndex++;
+        }
+
+        if (nameSuffixIndex == 0)
+        { // no colliders in the array - zero or one in total, the element member is used instead
+            AddCollider(link->collision, entityId, PrefabMakerUtils::MakeEntityName(link->name.c_str(), typeString), materialAsset);
+        }
+    }
+
+    void CollidersMaker::AddCollider(
+        urdf::CollisionSharedPtr collision,
+        AZ::EntityId entityId,
+        const AZStd::string& generatedName,
+        const AZ::Data::Asset<Physics::MaterialAsset>& materialAsset)
+    {
+        if (!collision)
+        { // it is ok not to have collision in a link
+            return;
+        }
+        AZ_TracePrintf(Internal::collidersMakerLoggingTag, "Processing collisions for entity id:%s\n", entityId.ToString().c_str());
+
+        auto geometry = collision->geometry;
+        if (!geometry)
+        { // non-empty visual should have a geometry
+            AZ_Warning(Internal::collidersMakerLoggingTag, false, "No Geometry for a collider of entity %s", entityId.ToString().c_str());
+            return;
+        }
+
+        AddColliderToEntity(collision, entityId, materialAsset);
+    }
+
+    void CollidersMaker::AddColliderToEntity(
+        urdf::CollisionSharedPtr collision, AZ::EntityId entityId, const AZ::Data::Asset<Physics::MaterialAsset>& materialAsset) const
+    {
+        // TODO - we are unable to set collider origin. Sub-entities don't work since they would need to parent visuals etc.
+        // TODO - solution: once Collider Component supports Cylinder Shape, switch to it from Shape Collider Component.
```

Comment:
Is this a Github issue for O3DE?
If so it should be created and the TODO replaced with a link to the issue.